src/avram/where.cr

```diff
@@ -30,7 +30,7 @@ module Avram::Where
   end
 
   abstract class ValueHoldingSqlClause < SqlClause
-    getter value : String | Array(String) | Array(Int32)
+    getter value : String | Array(String) | Array(Int32) | Nil
```

Comment:
This was needed in order to get everything to compile. Though, adding it may lead to other issues. This was also brought up in a previous comment https://github.com/luckyframework/avram/issues/27#issuecomment-486836365 But if we're going to treat `nil` as a `SuccessfulCast`, and treat empty strings as nil, then this just may be a side affect we deal with.

Reply:
Good news! `UserQuery.new.name(nil)` still raises a compile-time error saying to not do that. `UserQuery.new.where(:name, nil)` does work.